### ISSUE TITLE
ArangoDB High RAM Usage

### ISSUE THREAD

**Issue by cbluth** [arangodb/arangodb#10059]
## My Environment

* __ArangoDB Version__:        `3.4.6.1-1`
* __Storage Engine__:            `RocksDB`
* __Deployment Mode__:       `Cluster`
* __Deployment Strategy__:   `ArangoDB Starter`
* __Configuration__:               three nodes, baremetal
* __Infrastructure__:               baremetal
* __Operating System__:        `Ubuntu 18.04`
* __Total RAM in your machine__:  `64GiB`
* __Disks in use__:        ssd and hdd
* __Used Package__:              `Ubuntu.deb repo`

__Problem__:

I have a three node cluster, and it is having ram issues.
They are consuming way too much RAM when compared to its typical behavour.
I also have about 15GBs of logs on each node, here is a sample.

[arangodb.log](https://github.com/arangodb/arangodb/files/3641263/arangodb.log)

![image](https://user-images.githubusercontent.com/6709636/65411807-20937380-ddee-11e9-848b-800e34c34ee8.png)

How do I troubleshoot this issue?

**Comment by kvahed** [comment]
When you acquire `curl coordinator/_admin/cluster/shardDistribution` you can see, which collection is affected as there is a shard mapping for `production-db/29192308`. It seems as if we cannot complete the synchronisation for that shard. 
could you then raise the log levels for cluster and maintenance topics  on the spamming cluster machine like so:
`curl server:port/_admin/log/level -XPUT -d '{"cluster":"debug","maintenance":"debug"}'`
for just a couple of minutes and provide us with that log, please

**Comment by cbluth** [comment]
@kvahed 

here is the output from the first curl command:
```
root@db1:~# curl --basic --user "${DBUSER}:${DBPASS}" -s http://localhost:8529/_admin/cluster/shardDistribution | jq .
{
  "results": {
    "_statistics15": {
      "Plan": {
        "s3010038": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010038": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_statistics": {
      "Plan": {
        "s3010035": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010035": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_appbundles": {
      "Plan": {
        "s3010029": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010029": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_aqlfunctions": {
      "Plan": {
        "s3010016": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010016": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_iresearch_analyzers": {
      "Plan": {
        "s3010009": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010009": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_modules": {
      "Plan": {
        "s3010007": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010007": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_fishbowl": {
      "Plan": {
        "s11020022": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s11020022": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_users": {
      "Plan": {
        "s3010004": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010004": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_graphs": {
      "Plan": {
        "s3010002": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010002": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_routing": {
      "Plan": {
        "s3010011": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010011": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_statisticsRaw": {
      "Plan": {
        "s3010032": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010032": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_queues": {
      "Plan": {
        "s3010020": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010020": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_jobs": {
      "Plan": {
        "s3010022": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010022": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_frontend": {
      "Plan": {
        "s3010018": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010018": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    },
    "_apps": {
      "Plan": {
        "s3010026": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      },
      "Current": {
        "s3010026": {
          "leader": "DBServer0002",
          "followers": [
            "DBServer0003"
          ]
        }
      }
    }
  },
  "error": false,
  "code": 200
}
root@db1:~# 
```


^ im not quite sure what i should be looking for in here.

i have some hesitation about turning on the debugging logging without the knowledge on how to turn it back off.
@kvahed is there a way to query its current logging configuration so that i can change it back?

**Comment by cbluth** [comment]
@kvahed i was able to grab the logs with DEBUG turned on, see attached.

[arangod.log.gz](https://github.com/arangodb/arangodb/files/3641658/arangod.log.gz)

**Comment by dothebart** [comment]
As discussed in https://github.com/arangodb/arangodb/issues/11577 [exposing more options of rocksdb to the commandline](https://www.arangodb.com/docs/stable/programs-arangod-options.html#rocksdb) seems to ease the memory situation.

Namely:
 -   `--rocksdb.cache-index-and-filter-blocks-with-high-priority`
 -   `--rocksdb.pin-l0-filter-and-index-blocks-in-cache`
 -   `--rocksdb.pin-top-level-index-and-filter`

These have been added to [the now released ArangoDB 3.7](https://www.arangodb.com/2020/08/arangodb-3-7-a-big-step-forward-for-multi-model/) - If you still experience this, please get back to us in a new issue.
